ios/AppleAdsAttribution.m

```diff
@@ -115,9 +115,32 @@ + (NSString *) getAdServicesAttributionToken:(NSError * _Nullable *)error {
     {
         Class AAAttributionClass = NSClassFromString(@"AAAttribution");
         if (AAAttributionClass) {
-            NSString *attributionToken = [AAAttributionClass attributionTokenWithError:error];
-            if (*error == nil && attributionToken) {
-                return attributionToken;
+            // The code below is based on https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/pull/1114
+            // We are getting reports on iOS 14.5 that this API can hang, adding a short timeout for now.
+            dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+            __block NSString *token = nil;
+            __block NSError *innerError = nil;
+            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                token = [AAAttributionClass attributionTokenWithError:&innerError];
+                dispatch_semaphore_signal(semaphore);
+            });
+
+            // Apple said this API should respond within 50ms, lets give up after 100 ms
+            dispatch_semaphore_wait(semaphore, dispatch_time(DISPATCH_TIME_NOW, (int64_t)(100 * NSEC_PER_MSEC)));
```

Comment:
I notice that Branch.io uses 500ms as the timeout -- curious if there was a particular reason you switched to 100ms instead of 500ms